e2e-tests/cypress/e2e/journeys/insurance/your-business/company-details/company-details-continue-no-errors.spec.js

```diff
@@ -0,0 +1,56 @@
+import { companyDetails } from '../../../../pages/your-business';
+import { submitButton, yesRadioInput } from '../../../../pages/shared';
+import {
+  ROUTES, VALID_PHONE_NUMBERS, WEBSITE_EXAMPLES, COMPANIES_HOUSE_NUMBER,
+} from '../../../../../../constants';
+import getReferenceNumber from '../../../../helpers/get-reference-number';
+
+const natureOfBusinessUrl = `${Cypress.config('baseUrl')}${ROUTES.INSURANCE.EXPORTER_BUSINESS.NATURE_OF_BUSINESS}`;
+
+describe("Insurance - Your business - Company details page - As an Exporter I want to enter details about my business in 'your business' section", () => {
+  let referenceNumber;
+  let url;
+
+  before(() => {
+    cy.navigateToUrl(ROUTES.INSURANCE.START);
+
+    cy.submitInsuranceEligibilityAndStartApplication();
+
+    getReferenceNumber().then((id) => {
+      referenceNumber = id;
+
+      url = `${Cypress.config('baseUrl')}${ROUTES.INSURANCE.ROOT}/${referenceNumber}${ROUTES.INSURANCE.EXPORTER_BUSINESS.COMPANY_DETAILS}`;
+
+      cy.navigateToUrl(url);
+
+      cy.url().should('eq', url);
+    });
+  });
+
+  beforeEach(() => {
+    Cypress.Cookies.preserveOnce('_csrf');
+    Cypress.Cookies.preserveOnce('connect.sid');
+  });
+
+  describe('continue to next page', () => {
+    it('should not display any validation errors and redirect to the next page if required fields entered correctly', () => {
+      companyDetails.companiesHouseSearch().type(COMPANIES_HOUSE_NUMBER);
+      yesRadioInput().first().click();
+      yesRadioInput().eq(1).click();
+      submitButton().click();
+      cy.url().should('eq', natureOfBusinessUrl);
+    });
+
+    it('should not display any validation errors and redirect to the next page if required and optional fields entered correctly', () => {
+      cy.navigateToUrl(url);
+
+      companyDetails.companiesHouseSearch().type(COMPANIES_HOUSE_NUMBER);
+      yesRadioInput().first().click();
+      yesRadioInput().eq(1).click();
+      companyDetails.phoneNumber().clear().type(VALID_PHONE_NUMBERS.LANDLINE);
+      companyDetails.companyWebsite().clear().type(WEBSITE_EXAMPLES.VALID);
+      submitButton().click();
+      cy.url().should('eq', natureOfBusinessUrl);
+    });
+  });
```

Comment:
Do you think we should have the form submission test in the main `company-details` E2E test? This is what i've done for all policy and contract pages, check out `single-contract-policy.spec.js`, quite like this pattern and it saves some time running the E2E suite

```js
describe('form submission', () => {
  it(`should redirect to ${ABOUT_GOODS_OR_SERVICES}`, () => {
    // ...
  });
  describe('after submitting the form', () => {
    // ...
    describe('when going back to the page', () => {
      it('should have the submitted values', () => {
  });
```

Reply:
Also not for this PR, but when you get to the next part of the flow it's worth wrapping the code for "complete and submit this form" into a cypress command so we can easily complete and submit the form from anywhere, e.g

```js
cy.completeAndSubmitSingleContractPolicyForm();
```